src/distPlus/symbolic/MathJsParser.re

```diff
@@ -255,19 +325,28 @@ module MathAdtToDistDst = {
       | Object(_) => Error("Object not valid as top level")
     );
 
-  let run = (r): result(SymbolicDist.bigDist, string) =>
+  let run = (r): result(SymbolicDist.distTree, string) =>
     r |> MathAdtCleaner.run |> topLevel;
 };
 
 let fromString = str => {
+  /* We feed the user-typed string into Mathjs.parseMath,
+     which returns a JSON with (hopefully) a single-element array.
+     This array element is the top-level node of a nested-object tree
+     representing the functions/arguments/values/etc. in the string.
+
+     The function MathJsonToMathJsAdt then recursively unpacks this JSON into a typed data structure we can use.
+     Inside of this function, MathAdtToDistDst is called whenever a distribution function is encountered.
+   */
```

Comment:
Kudos for the comment!
Later on we may want to write our own parser for this. Mathjs' parser is kind of limited and not really made to be used just as a parser.